Timeline.cs

```diff
@@ -0,0 +1,506 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+/// <summary>
+/// A collection of <see cref="System.DateTime"/> and <see cref="TValue"/> sorted by <see cref="System.DateTime"/> field.
+/// </summary>
+/// <typeparam name="TValue">Value associated with a <see cref="System.DateTime"/>.</typeparam>
+internal class Timeline<TValue>:
+    IEnumerable<(DateTime Time, TValue Value)>,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['ICollection<(DateTime Time, TValue Value)>' implements 'IEnumerable<(DateTime Time, TValue Value)>' so 'IEnumerable<(DateTime Time, TValue Value)>' can be removed from the inheritance list.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5574777)